launcher/ui/pages/modplatform/modrinth/ModrinthPage.cpp

```diff
@@ -296,15 +296,14 @@ void ModrinthPage::suggestCurrent()
         return;
     }
 
-    for (auto& ver : current.versions) {
-        if (ver.id == selectedVersion) {
-            dialog->setSuggestedPack(current.name, ver.version, new InstanceImportTask(ver.download_url, this));
-            auto iconName = current.iconName;
-            m_model->getLogo(iconName, current.iconUrl.toString(),
-                             [this, iconName](QString logo) { dialog->setSuggestedIconFromFile(logo, iconName); });
-
-            break;
-        }
+    auto selectedVer_it = std::find_if(current.versions.cbegin(), current.versions.cend(), [this](const auto& ver) { return ver.id == selectedVersion; });
+    if (selectedVer_it != current.versions.cend())
+    {
+        Modrinth::ModpackVersion ver = *selectedVer_it;
```

Comment:
Are you sure you want to make a copy here? `ModpackVersion` seems pretty heavy with 6 `QString`s.